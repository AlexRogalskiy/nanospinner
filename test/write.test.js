let { createSpinner } = require('../index.js')

let stdout = { out: '' }
stdout.write = (symbols) => {
  stdout.out += symbols
}

it(`marks spinner as write`, () => {
  stdout.out = ''
  let spinner = createSpinner('#write', { stream: stdout })

  spinner.write('Write text')

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})

it(`marks spinner as write whit clear`, () => {
  stdout.out = ''
  let spinner = createSpinner('#write', { stream: stdout })

  spinner.write('Write text', true)

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})
