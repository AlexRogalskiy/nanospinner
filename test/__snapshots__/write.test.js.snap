// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marks spinner as write whit clear: CI 1`] = `"Write text"`;

exports[`marks spinner as write whit clear: Local 1`] = `"[1GWrite text"`;

exports[`marks spinner as write: CI 1`] = `"Write text"`;

exports[`marks spinner as write: Local 1`] = `"Write text"`;
